src/cargo/ops/registry.rs

```diff
@@ -35,11 +35,17 @@ pub struct PublishOpts<'cfg> {
     pub index: Option<String>,
     pub verify: bool,
     pub allow_dirty: bool,
+    pub dry_run: bool,
 }
 
 pub fn publish(ws: &Workspace, opts: &PublishOpts) -> CargoResult<()> {
     let pkg = try!(ws.current());
 
+    if !opts.verify && opts.dry_run {
+        bail!("cannot specify both no verify and dry run \
+               simultaneously");
+    }
```

Comment:
It was in the original PR. My guess is that it's not a full dry run if the verification step isn't run. Combining --no-verify and --dry-run works fine with the guard removed so I can remove it if you like.